src/cplscheme/CompositionalCouplingScheme.cpp

```diff
@@ -201,12 +201,24 @@ bool CompositionalCouplingScheme::hasTimeWindowSize() const
   return hasIt;
 }
 
+bool CompositionalCouplingScheme::solverSetsTimeWindowSize() const
+{
+  PRECICE_TRACE();
+  bool setsIt = false;
+  for (const Scheme &scheme : _couplingSchemes) {
+    setsIt &= scheme.scheme->solverSetsTimeWindowSize();
```

Comment:
```suggestion
    setsIt |= scheme.scheme->solverSetsTimeWindowSize();
```
?

Reply:
Yes, makes sense now as it's just the negative of the old `hasTimeWindowSize`.

Reply:
Actually... then it should be `&=` for `solverSetsTimeWindowSize`. That's simply [De Morgan's law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). I just reverted the latest commit via [e0f9306](https://github.com/precice/precice/pull/1307/commits/e0f9306823a94be955e922e42b0849d058c0d25a).

Reply:
To conclude this discussion: The both functions are not inverse of each other (otherwise solver A sending timestep size to B and B forwarding to C would not work at all). This counters the De Morgan argument. `|=` is the correct implementation, because a compositional coupling scheme might have only a single scheme that sends the timestep size. However: I decided to entirely remove this function (see below). So this explanation is mainly for historic reasons.

Reply:
To conclude this discussion: The both functions are not inverse of each other (otherwise solver A sending timestep size to B and B forwarding to C would not work at all). This counters the De Morgan argument. `|=` is the correct implementation, because a compositional coupling scheme might have only a single scheme that sends the timestep size. However: I decided to entirely remove this function (see below). So this explanation is mainly for historic reasons.